### DISCUSSION TITLE
Sync doesn't work with pCloud (WebDAV)

### DISCUSSION THREAD

**Discussion by Juelicher-Trainee** [remotely-save/remotely-save#661] [Q&A] [answered]
Hi everyone,

I am new to this plugin (and new to Obsidian). I set up the plugin to use pCloud WebDAV interface. I used "Check" to verify the connection. All OK.

I can start sync, but after "Preparing" and Syncing and then counting, e.g. "1/2891" it breaks off. Mostly immediately, sometimes after having reached 24/2891, or so.

I looked into the directory on pCloud. Since I am using encryption, encrypted files appear on in the "Obsidian Vault2 folder. Most of them have site 0 bytes, but some are larger.

I set auto-syncing and I have now a slowly growing number of files in the folder on pCloud, but sync breaks off with "Never synced") every time.

Can anyone help? Where do I start, what parameters can I change?
Any help appreciated.

J.T.

**Comment by fyears** [comment]
> I can start sync, but after "Preparing" and Syncing and then counting, e.g. "1/2891" it breaks off.

any error hints?

**Reply by Juelicher-Trainee** [reply]
I turned on debug, I get messages on the screen (but cannot find the log file on Windows :-( )
It seems to connect and start syncing. Then a few notes are shown by title and the message "Request failed, status 500".
After a few of these "too many errors".

Thank you.

**Reply by fyears** [reply]
so pcloud doesn't allow connecting or you're using wrong configuration. please check your settings firstly. for example, try the address with or without trailing slash or not

**Reply by Juelicher-Trainee** [reply]
Thank you. I am in a mess now.

- I realized that the Vault was on a iCloud drive on Windows. I thought this was a problem and moved it out of the iCloud drive, connected to the new location and tried to sync. 
- It didn't work at all, so I thought I reinstall the plugin to clean it up
- Ever since I cannot connect (previously I could connected and some 800 small encrypted filechunks were actually transferred)
- NOw I am getting "401" request failed, unauthorized.
- I tried both "https://ewebdav.pcloud.com/" and "https://ewebdav.pcloud.com"
- The only change I made to pcloud is that I have turned on 2FA with an OTP code. That surely cannot prevent login, or can it?
Thank you.

**Reply by fyears** [reply]
i don't know, i don have pcloud account, and pcloud doesn't provide webdav access for free account.

i don't think the plugin cares the vault on icloud or not, and i strongly recommend you NOT mox up two sync methods.

> The only change I made to pcloud is that I have turned on 2FA with an OTP code. That surely cannot prevent login, or can it?

i don't know, how about disable it and try again?

**Comment by Aviator85** [comment]
I'm having sync issues using SafeInCloud - have used pcloud and SafeInCloud for a few years no problems but havent been able to sync. Must be a webdav issue? 

Error below, i chopped some info for security 

Error Domain=sync_error Code=1 "(null)" UserInfo={NSUnderlyingError=0x6000001228b0 {Error Domain=com.alamofire.error.serialization.response Code=-1011 "Request failed: forbidden (403)" UserInfo={NSLocalizedDescription=Request failed: forbidden (403), statusCode=403, NSErrorFailingURLKey=https://webdav.pcloud.com/SafeInCloud.db, com.alamofire.serialization.response.error.data={length = 319, bytes = }, com.alamofire.serialization.response.error.response=<NSHTTPURLResponse: > { URL: https://webdav.pcloud.com/SafeInCloud.db } { Status Code: 403, Headers {
    Connection =     (
        "Keep-Alive"
    );
    "Content-Length" =     (
        319
    );
    "Content-Type" =     (
        "text/html; charset=iso-8859-1"
    );
    Date =     (
        "Thu, 16 May 2024 21:44:01 GMT"
    );
    "Keep-Alive" =     (
        "timeout=5, max=100"
    );
    Server =     (
        "Apache/2.4.10 (Debian)"
    );
} }}}}

**Reply by fyears** [reply]
do you mean your safeincloud app cannot sync either now but can sync a while before? if you haven't update the app then that should be a problem with webdav

**Reply by Aviator85** [reply]
Seemed to just be that the database file was corrupted - i overwrote the file with a backup that was made a few days earlier and syncs just fine now....

**Comment by Juelicher-Trainee** [comment]
Hello everyone, 

the OP here again. I found out the hard way that 2FA must be switched off for pCloud to use the webdav interface.

Alas, that did nit help me much. I can connect now to pcloud but sync breaks off after a few (encrypted) folders are created on pcloud. I get a bunch of error messages, but they don't mean much to me, as I am not skilled enough.

Would anyone care to look at them?

Thank you very much.

Obsidian Developer Console
plugin:table-editor-obsidian:28854 loading markdown-table-editor plugin
plugin:highlightr-plugin:3508 Highlightr v1.2.2 loaded
plugin:remotely-save:335 loading plugin remotely-save
plugin:remotely-save:335 db connected
2plugin:remotely-save:330 remote vault folder not exists, creating
plugin:remotely-save:330 remote vault folder created!
plugin:remotely-save:330 Error: Invalid response: 405 Method Not Allowed
    at Le (plugin:remotely-save:330:261430)
    at Ue (plugin:remotely-save:330:261694)
    at eval (plugin:remotely-save:330:268286)
eval @ plugin:remotely-save:330
plugin:remotely-save:335 startting sync.
plugin:remotely-save:335 Error: Invalid response: 403 Forbidden
    at Le (plugin:remotely-save:330:261430)
    at Ue (plugin:remotely-save:330:261694)
    at eval (plugin:remotely-save:330:270906)
eval @ plugin:remotely-save:335
plugin:remotely-save:335 endding sync.
plugin:remotely-save:335 startting sync.
plugin:remotely-save:335 AggregateError: 
    Error: $Import$/Attachments/: net::ERR_CONNECTION_TIMED_OUT
        at eval (plugin:remotely-save:335:233825)
    at eval (plugin:remotely-save:335:233985)
    at Generator.next (<anonymous>)
    at s (plugin:remotely-save:181:296564)
eval @ plugin:remotely-save:335
plugin:remotely-save:335 endding sync.
plugin:remotely-save:335 startting sync.
plugin:remotely-save:335 Error: net::ERR_NETWORK_IO_SUSPENDED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:108519)
    at SimpleURLLoaderWrapper.emit (node:events:517:28)
eval @ plugin:remotely-save:335
plugin:remotely-save:335 endding sync.

**Comment by fyears** [comment]
net::ERR_NETWORK_IO_SUSPENDED

it seems that your network is lost. you can retry again?

**Reply by Juelicher-Trainee** [reply]
I am still struggling with this problem. I did try to repair the networking part of Windows using the tutorial on the web.
Still, soon after I start the sync I get in the Obsidian Developer Console:

`plugin:remotely-save:335 Error: net::ERR_CONNECTION_TIMED_OUT
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:108519)
    at SimpleURLLoaderWrapper.emit (node:events:517:28)
`

and after that soon an "Aggregate Error" and many instances of 

`Error: MyFolder/MyNote.md: Invalid response: 500 Internal Server Error at eval (plugin:remotely-save:335:233825)`

Now I am totally lost.

Other apps (e.g Evernote that does quite a bit of syncing, too) seem to work just fine on the same laptop.

How could I start to track down the issue?

Any help appreciated, thank you.

**Reply by fyears** [reply]
it's still likely to be a configuration error. what's your connection address? have you disabled the 2FA as other user said?

**Reply by Juelicher-Trainee** [reply]
I connect to https://ewebdav.pcoud.com, as I am in European domain. (I hope that's what you mean.)
I did disable 2FA, before that I could not even connect to pcloud.
Now I can and the Vault is about 3 MB with ≈ 600 files on pcloud (encrypted). The local Vault is ≈ 900 MB with ≈ 4000 files.
So there is some syncing going on.

Thank you for your help.

**Comment by Juelicher-Trainee** [comment]
As I am desperate, I tried the webdav service of another provider. It seems to work without network problems.
Sync doesn't stop, no error reports. On my ios mobile device, it sort of works too.

What doesn't: 

1.  I couldn't make the QR  or URI import work
2. If I want to enter the encryption pass phrase, the mask (rather wide when showing only asterix) shrinks to 1 charcter wide or less. It is possible to paste the password into it (presumably), but not possible to check it if correct (mask is very narrow). This is on an iPad mini.
3. I don't seem to be able to sync attachments. Not even PDF-files are synced to the mobile. The size of the vault is very small, since it contains the notes only.

But these are not pcloud issues, I realize.

**Reply by fyears** [reply]
1. Are you using the same vault name? Import only works with same vault name. And what's the error?
2. There is an eye icon beside the password input field. You can click it to view the password.
3. Pdf not syncing is quite strange. In fact the plugin doesn't recognize pdf files or any other files: it just syncs whatever files it sees. And errors? Have you set skipping large files or set the deny list? Does the pdf file show up on the remote?

**Reply by Juelicher-Trainee** [reply]
1)
I am using the same Vault name. It says OK, I should restart the plugin. I did, it didn't help. But I did have the old config in the masks, is that perhaps the problem? Nothing has been rewritten. 

2)
That is exactly the problem. The mask with asterix' has a width of ca. 8-10 characters. When I tap the eye, the field shrinks to perhaps just one character width, or less. It is very narrow, I can see nothing in it. Even if I turn the ipad to landscape, the mask is very narrow, but the explanatory text to the left of it becomes wider. 

3)
Skip large files is deactivated, but I also tried 500MB, which I don't have. Doesn't help. It is not just PDFS. The vault is 4MB on the WEBdav server. But the original one on the laptop is 800 MB or so. It seems that attachments are not synched to the server. I'll try it without encryption, perhaps then we can see what's missing. 

Thanks a million for your availability and help.

**Comment by fyears** [comment, marked as answer]
hi i have implemented a pcloud PRO feature with its native api instead of webdav. you are welcome to give it a try.

**Reply by ghdragunoff** [reply]
Hi! Where can I find information about connecting PRO? Features are not available after October 31st 😫

**Reply by fyears** [reply]
see #879

**Comment by heatfighter** [comment]
Hi,
I have very similar problems with the pCloud WebDAV sync (Europe servers, 2FA off, lifetime account). My database is only ~40 MB and it took me ~10 manual syncs to seemingly get all the data to pCloud because I got all kinds of error message (just like Juelicher-Trainee described). Every sync run it transferred a bit more. Now it claims everything is in sync. However, even if there are no file changes and hence nothing to really transfer, the (successful) syncs takes ~40 s for this relatively small database!
I tried the same database with the Cloud service Koofr.net (free account), which also offers WebDAV and is also located in Europe. Here the sync without file transfer is done in ~4 s.
Does anybody have an idea, why the pCloud WebDAV access is so slow?

**Reply by fyears** [reply]
i don't know, because i don't have pcloud paid account to test webdav. but actually you can try pcloud native pro sync in remotely save and see the time difference

**Reply by heatfighter** [reply]
Hi fyears,
first of all, I really appreciate your efforts to write this plugin, which works very well for me in general. I especially appreciate the feature to end-to-end encrypt the database, so I don't have to trust my cloud service, no monkeying with Cryptomator or such. Very nice. 

I understand that I can use the pCloud native pro sync feature (which I tried, it is indeed much faster, very much comparable to Koofr via WebDAV) and I understand that you have an interest in recommending people with problems with their WebDAV access the pro feature :-) If there would be an option to buy and not rent the pro features I would do it immediateyl. However, I chose pCloud also because it offers a WebDAV interface, and I already paid for that cloud service including the WebDAV interface (lifetime plan). So, I really would like to use it and I think some other people might be in the same situation. That's why I asked whether someone has an idea why the pCloud WebDAV interface is so slow. I don't think it really is the plugin, the Koofr WebDAV interface works just fine, but maybe it is a certain WebDAV command that creates problems with the pCloud WebDAV interface implementation...?
Thanks again.

**Reply by Juelicher-Trainee** [reply]
Even though I cannot help, I just wanted to remark that I am in the exact same situation.

Since the monthly subscriptions for all kinds of stuff are going through the roof, I wanted to keep just the essentials. 

Of course, we don't know yet, if this plugin-will be expensive in the PRO version and whether a lifetime deal will be offered. I certainly hope so.

Having said that, I am happy with how well the pCloud native version works, good job fyears!
